functions.py

```diff
@@ -2,17 +2,17 @@
 
```

Comment:
is there a better name for this file than functions? I guess it kind of is a trash bin of random stuff.

Reply:
if we decompose it, certainly. the new names will be obvious.